generate-types.js

```diff
@@ -0,0 +1,53 @@
+const TypeWriter = require('@gimenete/type-writer')
+const webhooks = require('@octokit/webhooks-definitions')
+const pascalCase = require('pascal-case')
+const prettier = require('prettier')
+const fs = require('fs')
+
+const signatures = []
+const tw = new TypeWriter()
+for (const webhook of webhooks) {
+  const { name, actions, examples } = webhook
+  if (!examples) continue
+  const typeName = `WebhookPayload${pascalCase(name)}`
+  tw.add(examples, { rootTypeName: typeName, namedKeyPaths: { [`${typeName}.repository`]: 'PayloadRepository' } })
+  const events = [
+    `'${name}'`,
+    ...actions.map(action => `'${name}.${action}'`)
+  ].join(' | ')
+  signatures.push(`
+    public on (event: ${events}, callback: (event: Webhooks.WebhookEvent<${typeName}>) => void): void
+  `)
+}
+
+const definition = `
+export interface Options {
+  path: string,
+  secret: string
+}
+
+${tw.generate('typescript', { inlined: false }).replace(/type /g, 'export type ')}
+
+export default class Webhooks {
+  constructor (Options)
+
+  public on (event: 'error', callback: (event: Error) => void): void
+  public on (event: '*' | string[], callback: (event: Webhooks.WebhookEvent<any> | Error) => void): void
+  ${signatures.join('\n')}
+
+  public sign (data: any): string
+}
+
+export interface WebhookEvent<T> {
```

Comment:
Also, this needs to be moved into a namespace
```suggestion
export namespace Webhooks {
    export interface WebhookEvent<T> {
```